src/uneditable-section/section-item.tsx

```diff
@@ -1,17 +1,166 @@
+    onMask,
+    onUnmask,
+    onTryAgain,
+}: Props) => {
+    // =========================================================================
+    // CONST, STATE, REF
+    // =========================================================================
+    const [displayMaskState, setMaskState] = useState<
+        UneditableSectionItemMaskState | undefined
+    >(maskState);
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        setMaskState(maskState);
+    }, [maskState]);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleClickableClick = () => {
+        if (maskLoadingState === "fail") {
+            if (onTryAgain) onTryAgain();
+        }
+
+        switch (displayMaskState) {
+            case "masked": {
+                if (onUnmask) {
+                    onUnmask();
+                }
+                setMaskState("unmasked");
+                break;
+            }
+            case "unmasked": {
+                if (onMask) {
+                    onMask();
+                }
+                setMaskState("masked");
+                break;
+            }
+        }
+    };
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const getMaskedValue = () => {
+        return StringHelper.maskValue(value, {
+            maskChar,
+            maskRange,
+            unmaskRange,
+            maskRegex,
+            maskTransformer,
+        });
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderMaskingState = () => {
+        switch (maskLoadingState) {
+            case "fail":
+                return (
+                    <>
+                        <ErrorIcon />
+                        <ErrorLabel>Error</ErrorLabel>
+                        <TryAgainLabel weight="semibold">
+                            Try again?
+                        </TryAgainLabel>
+                    </>
+                );
+            case "loading":
+                return (
+                    <>
+                        <Spinner />
+                        <LoadingLabel>Retrieving...</LoadingLabel>
+                    </>
+                );
+            default:
+                return (
+                    <>
+                        <Text.Body>
+                            {displayMaskState === "masked"
+                                ? getMaskedValue()
+                                : value}
+                        </Text.Body>
```

Comment:
this would render a p tag in the button which is not recommended, could cast to span or apply the text styling on the button